executions/graphql-kotlin-transaction-loader-instrumentation/src/main/kotlin/com/expediagroup/graphql/transactionbatcher/instrumentation/syncexhaustion/state/ExecutionBatchState.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2022 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.transactionbatcher.instrumentation.syncexhaustion.state
+
+import graphql.language.Field
+import graphql.execution.ResultPath
+import graphql.schema.GraphQLType
+import graphql.schema.GraphQLTypeUtil.isList
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.ConcurrentHashMap
+
+class ExecutionBatchState {
+
+    private val executionStrategiesState: ConcurrentHashMap<String, ExecutionStrategyState> = ConcurrentHashMap()
+
+    fun addExecutionStrategyState(
+        field: Field?,
```

Comment:
is it possible to have nullable field?

Reply:
yes, the executionStrategy for root field is null